### PR TITLE
Ensure the temp directory exists

### LINKED ISSUE #41919: Check for the existence of Poifiles in temporary directory before creating XLSX files
### Describe the bug

Had a customer run into an issue where they were getting the following errors when attempting to download results as an XLSX file.

> ERROR async.streaming-response :: Caught unexpected Exception in streaming response body 
> 
> metabase: java.lang.RuntimeException: java.nio.file.NoSuchFileException: /tmp/poifiles/poi-sxssf-sheet5437265569215382033.xml 
> 
> metabase: at org.apache.poi.xssf.streaming.SXSSFWorkbook.createAndRegisterSXSSFSheet(SXSSFWorkbook.java:728)
> 

They’re hosted in an AWS EC2 Linux instance that periodically runs **systemd-tmpfiles --clean** to clean everything within the /tmp directory.
What happens is Metabase can’t locate **'/tmp/poifiles/'** as it’s going through the process of creating the XLSX file.
I mention '/tmp/poifiles/' as an example but it’s wherever the **java.io.tmpdir** directory is pointing to, which can be found by running the following command.

`java -XshowSettings:properties -version`

In this particular situation, I did suggest creating an exclusion in the cleanup process so the 'poifiles' folder is not removed during cleanup.
However, as a precautionary measure, should we be checking if that directory exists before attempting to write to it?

### To Reproduce

1. Run Metabase as you normally would.

2. Test downloading results of a card an XLSX to confirm it's working

3. Identify the location of the 'poifiles' temp folder. Either programmatically or running the following command **java -XshowSettings:properties -version** and locating the value for **java.io.tmpdir** 

4. Remove the 'poifiles' folder

5. Download results of a card to XLSX and errors should occur as 'poifiles' can't be located

6. Restarting or manually creating 'poifiles' resolves the issue

### Expected behavior

Possibly checking if 'poifiles' folder exists before creating XLSX files

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Mac OS X",
    "os.version": "13.5.1",
    "user.language": "en",
    "user.timezone": "America/Chicago"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "mongo",
      "bigquery-cloud-sdk",
      "mysql",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "15.3"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-04-16",
      "tag": "v1.49.6",
      "hash": "5abf130"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

### Severity

P3 - There are work arounds and even restarting Metabase should resolve the issue.  However, a suggestion for improving customer experience.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dpsutton**
Fixes #41919

tl;dr: apache poi keeps some scratch documents on disk to make workbooks. It keeps them (tidily) in a directory at `<tempfile-location>/poifiles`. But if that temp file goes away it isn't smart enough to ensure it does exist. 

This has been fixed in their repo but not yet released. It will be in 5.3.1 when it comes out. They have ~ a yearly cadence of releases so might be a while. Let's ensure the file exists. The test can stay when we bump to 5.3.1 and we can remove our check to ensure the directory exists.

https://github.com/metabase/metabase/issues/41919#issuecomment-2400542908

This issue goes away in 5.3.1 of apache poi. Seems like they have a yearly release cadence so we can ensure this exists for the time being and then remove this when we can bump to 5.3.1

```diff
index b763a1ffdf..c87992c935 100644
--- a/deps.edn
+++ b/deps.edn
@@ -122,7 +122,7 @@
   {:mvn/version "2.23.1"}             ; allows the slf4j2 API to work with log4j 2
   org.apache.logging.log4j/log4j-layout-template-json
   {:mvn/version "2.23.1"}             ; allows the custom json logging format
-  org.apache.poi/poi                        {:mvn/version "5.2.5"}              ; Work with Office documents (e.g. Excel spreadsheets) -- newer version than one specified by Docjure
+  org.apache.poi/poi                        {:mvn/version "5.3.1"}              ; Work with Office documents (e.g. Excel spreadsheets) -- newer version than one specified by Docjure
   org.apache.poi/poi-ooxml                  {:mvn/version "5.2.5"
                                              :exclusions  [org.bouncycastle/bcpkix-jdk15on
                                                            org.bouncycastle/bcprov-jdk15on]}
```

### CODE REVIEW

**Review by adam-james-v — APPROVED** [review]

**Comment by dpsutton** [comment]
@metabase-bot backport release-x.50.x
